lib/curator.rb

```diff
@@ -0,0 +1,85 @@
+require './lib/fileio'
+
+class Curator
+  attr_reader :photographs, :artists
+
+  def initialize
+    @photographs = []
+    @artists = []
+  end
+
+  def add_photograph(photo)
+    @photographs << photo
+  end
+
+  def add_artist(artist)
+    @artists << artist
+  end
+
+  def find_artist_by_id(id)
+    @artists.find { |artist| artist.id == id }
+  end
+
+  def find_photograph_by_id(id)
+    @photographs.find { |photo| photo.id == id }
+  end
+
+  def find_photographs_by_artist(artist)
+    @photographs.find_all { |photo| photo.artist_id == artist.id }
+  end
+
+  def count_of_photos_by_artist_id
+    @photographs.inject(Hash.new(0)) do |photos_by_artist, photo|
+      photos_by_artist[photo.artist_id] += 1
+      photos_by_artist
+    end
+  end
+
+  def artists_ids_with_multiple_photos
+    count_of_photos_by_artist_id.select do |artist_id, photo_num|
+      photo_num > 1
+    end
+  end
+
+  def artists_with_multiple_photographs
+    artist_ids = artists_ids_with_multiple_photos.keys
+    found_artists = []
+    artist_ids.each do |id|
+      @artists.each do |artist|
+        found_artists << artist if artist.id == id
+      end
+    end
+    found_artists
+  end
```

Comment:
good job here, but I think you were overthinking it a little; try to reuse methods you've already written as much as possible:

```ruby
  def artists_with_multiple_photographs
    @artists.find_all do |artist|
      find_photographs_by_artist(artist).length > 1
    end
  end
```